Rostislav korst hw3 java bit 2020

RostislavKorst opened the pull request.

RostislavKorst committed:
Add files via upload

RostislavKorst committed:
Add files via upload

RostislavKorst committed:
hw3 first commit

RostislavKorst committed:
Исправление ошибок после ревью по дз2

vladkats commented on hw2/Account.java:
```diff
@@ -0,0 +1,127 @@
+package ru.sbt.mipt.hw2;
```
код и тесты в одном пакете, структура некорректная, и почему-то здесь оказалось дз2.

vladkats commented on hw3/src/ru/sbt/mipt/hw3/DebitCard.java:
```diff
@@ -0,0 +1,160 @@
+    private final TransactionManager transactionManager;
+    private final Entries entries;
+    private static long counter = 1;
+    private final double bonusAccountPercent;
```
здесь этого не должно быть. Эта информация должна быть только в BonusAccount

vladkats commented on hw3/src/ru/sbt/mipt/hw3/DebitCard.java:
```diff
@@ -0,0 +1,160 @@
+        if (amount > 0 && (currentBalance - amount) >= 0) {
+            Transaction transaction = transactionManager.createTransaction(amount, this, beneficiary);
+            transactionManager.executeTransaction(transaction);
+            addPointsToBonusAccount(amount, transaction);
```
добавление бонусов должно быть в одной транзакции со всем остальным. Если транзакция не успешна, то бонусы не должны прибавляться

vladkats commented on hw3/src/ru/sbt/mipt/hw3/AnalyticsManagerTest.java:
```diff
@@ -0,0 +1,240 @@
+        assertEquals(returnedAccount, beneficiary2);
+    }
+
+    private void executeSomeTransactionsWithMostFrequentBeneficiary2() {
```
private методы внизу. Они не должны отвлекать от самих тестов

vladkats commented on hw3/src/ru/sbt/mipt/hw3/SimpleEntitiesStorageTest.java:
```diff
@@ -0,0 +1,161 @@
+        debitCard1.withdraw(150, debitCard2);
+        debitCard2.withdraw(400, debitCard1);
+        //when
+        storage.save(debitCard1);
+        storage.save(debitCard2);
+        storage.save(debitCard3);
+        storage.save(debitCard4);
+        storage.save(debitCard5);
+        storage.save(debitCard6);
```
это все подготовка данных, вынести в отдельный метод с понятным названием. 
А проверяешь ты вот это, оно и должно быть в when
storage.findAll()

vladkats commented on hw3/src/ru/sbt/mipt/hw3/TransactionManager.java:
```diff
@@ -0,0 +1,63 @@
+    }
+
+    public void rollbackTransaction(Transaction transaction) {
+        Transaction new_transaction = transaction.rollback();
```
new_transaction -> newTransaction

vladkats reviewed (commented):
Есть нюансы, связанные с добавлением бонусного аккаунта. Могу принять, но лучше бы переделать

RostislavKorst committed:
Исправление структуры директорий

RostislavKorst committed:
Убран bonusAccountPercent из DebitCard + Рефакторинг тестов

bonusAccountPercent - это сущность BonusAccount, и в DebitCard его быть не должно.

RostislavKorst committed:
l Please enter the commit message for your changes. Lines starting
